src/node/full_node.cpp

```diff
@@ -204,53 +233,22 @@ void FullNode::start() {
     started_ = false;
     return;
   }
-  if (jsonrpc_io_ctx_) {
-    if (jsonrpc_http_) {
-      jsonrpc_http_->StartListening();
-    }
-    if (jsonrpc_ws_) {
-      jsonrpc_ws_->run();
-      trx_mgr_->setWsServer(jsonrpc_ws_);
-      executor_->setWSServer(jsonrpc_ws_);
-    }
-
-    for (size_t i = 0; i < conf_.rpc->threads_num; ++i) {
-      jsonrpc_threads_.emplace_back([this] { jsonrpc_io_ctx_->run(); });
-    }
-  }
   started_ = true;
   LOG(log_nf_) << "Node started ... ";
-}  // namespace taraxa
+}
 
 void FullNode::close() {
-  util::ExitStack finally;
-  // because `this` shared ptr is given to it
-  finally += [this] { jsonrpc_api_.reset(); };
   if (bool b = false; !stopped_.compare_exchange_strong(b, !b)) {
     return;
   }
+  jsonrpc_api_ = nullptr;  // TODO Because it indirectly refers to FullNode
```

Comment:
TODO what ?

Reply:
I will make it more clear in the comment.
TODO - remove that line. We should not care about explicitly controlling destruction of objects. The lifecycle of objects should be as declarative as possible (kind of RAII).

Reply:
yeah this whole close() function might be in destructor I guess without the need to manually reset shared ptrs, although the author maybe implemented it this way for some reason...

Reply:
Improved the comment, please resolve